File: testdata/precision-suite-lua/clean_placeholders.lua
-- Clean: placeholder / example credentials. These are documentation defaults,
-- not real secrets — a scanner that flags them as hardcoded credentials is
-- producing a false positive. Mirrors the placeholder allowlist the other
-- language suites exercise.

local config = {
  api_key = "your-api-key-here",
  db_password = "changeme",
  smtp_password = "<your-smtp-password>",
  stripe_key = "sk_test_0000000000000000000000000",
  database_url = "postgres://USER:PASSWORD@localhost:5432/app",
}

-- os.getenv is the CORRECT way to read a secret at runtime; reading it is not a
-- vulnerability, and the fallback here is an obvious placeholder.
local token = os.getenv("API_TOKEN") or "REPLACE_ME_AT_DEPLOY_TIME"

return config, token

File: testdata/precision-suite-clojure/clean_placeholders.clj
(ns app.placeholders
  "Clean: placeholder / example credentials and env-var reads. These deliberately
   contain secret-shaped strings that must NOT be flagged (they are placeholders
   or come from the environment, not hardcoded live secrets). Any finding here is
   a false positive.")

;; Placeholder credentials — the example/placeholder allowlist must drop these.
(def config
  {:api-key   "your-api-key-here"
   :password  "changeme"
   :db-url    "postgres://USER:PASSWORD@localhost:5432/app"
   :smtp-pass "<your-smtp-password>"
   :token     "sk_test_00000000000000000000000000"})

;; Secrets pulled from the environment are not hardcoded.
(def runtime-key (System/getenv "APP_API_KEY"))
(def runtime-db  (System/getenv "DATABASE_URL"))
